src/core/sock.c

```diff
@@ -242,9 +243,51 @@ int nn_sock_term (struct nn_sock *self)
     /*  At this point, we can be reasonably certain that no other thread
         has any references to the socket. */
 
+    nn_sock_unsafe_cleanup(self, NN_CLEAN_DEFAULT);
+
+    return 0;
+}
+
+void nn_release_endpoint (struct nn_list_item *it)
+{
+    struct nn_ep *ep = nn_cont (it, struct nn_ep, item);
+    /* TODO: force cleanup the endpoint */
```

Comment:
This needs a hook inside the endpoint to release all of its resources at once without caring about safely terminating.